pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -257,8 +258,14 @@ func (p *Pod) Stop(ctx context.Context, c client.Client, _ []podset.PodSetInfo,
 		podsInGroup = []corev1.Pod{p.pod}
 	}
 
+	// Pods shouldn't be stopped if matching workload is not found,
+	// since stop will delete the pods.
+	if _, finished := p.Finished(); stopReason == jobframework.StopReasonNoMatchingWorkload && !finished {
+		return false, nil
+	}
```

Comment:
Let's remove this for now and focus on the case where the Workload was explicitly deleted.